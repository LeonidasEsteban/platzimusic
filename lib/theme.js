const theme = {
  font: {
    title: 'Quicksand, Helvetica, Arial',
  },
  flexboxgrid: {
    gutterWidth: 0,
    container: {
      md: 64,
      lg: 64,
    },
  },
  color: {
    grayA: '#e1e4ef',
    grayB: '#a6acc1',
    grayC: '#4b4e5a',
    blue: '#5179ff',
    pink: '#ea83ee',
  },
};

export default theme;

