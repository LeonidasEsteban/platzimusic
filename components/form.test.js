/* global it, expect, describe */
import React from 'react';
import { render } from 'react-dom';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Form from './form';
import theme from '../lib/theme';

function handleSubmit() {}

describe('Test Form component', () => {
  it('should render without default value', () => {
    const div = document.createElement('div');
    render(
      <ThemeProvider theme={theme}>
        <Form onSubmit={handleSubmit} />
      </ThemeProvider>,
      div,
    );
  });

  it('should render with default value', () => {
    const div = document.createElement('div');
    render(
      <ThemeProvider theme={theme}>
        <Form defaultValue="Testing" onSubmit={handleSubmit} />
      </ThemeProvider>,
      div,
    );
  });

  it('should render without default value the expected HTML', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Form onSubmit={handleSubmit} />
      </ThemeProvider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with default value the expected HTML', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Form defaultValue="Testing" onSubmit={handleSubmit} />
      </ThemeProvider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
