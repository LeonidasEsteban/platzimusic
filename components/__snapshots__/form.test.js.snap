// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Form component should render with default value the expected HTML 1`] = `
<form
  action="/results"
  method="GET"
  onSubmit={[Function]}
>
  <div
    className="eEhaii"
  >
    <div
      className="gTtEZJ"
    >
      <input
        autoFocus={true}
        className="hQNAuW"
        defaultValue="Testing"
        name="query"
        placeholder="Busca por canción, artista o album"
        type="search"
      />
    </div>
    <div
      className="gOMjZh"
    >
      <button
        className="hSBnIA"
      >
        Buscar
      </button>
    </div>
  </div>
</form>
`;

exports[`Test Form component should render without default value the expected HTML 1`] = `
<form
  action="/results"
  method="GET"
  onSubmit={[Function]}
>
  <div
    className="eEhaii"
  >
    <div
      className="gTtEZJ"
    >
      <input
        autoFocus={true}
        className="hQNAuW"
        defaultValue=""
        name="query"
        placeholder="Busca por canción, artista o album"
        type="search"
      />
    </div>
    <div
      className="gOMjZh"
    >
      <button
        className="hSBnIA"
      >
        Buscar
      </button>
    </div>
  </div>
</form>
`;
